master/buildbot/steps/source/git.py

```diff
@@ -422,21 +426,23 @@ def _fetchOrFallback(self, _=None):
     def _clone(self, shallowClone):
         """Retry if clone failed"""
 
-        args = []
+        command = ['clone']
         switchToBranch = False
         if self.supportsBranch and self.branch != 'HEAD':
             if self.branch.startswith('refs/'):
                 # we can't choose this branch from 'git clone' directly; we
                 # must do so after the clone
                 switchToBranch = True
-                args += ['--no-checkout']
+                command += ['--no-checkout']
             else:
-                args += ['--branch', self.branch]
+                command += ['--branch', self.branch]
         if shallowClone:
-            args += ['--depth', '1']
+            command += ['--depth', '1']
         if self.reference:
-            args += ['--reference', self.reference]
-        command = ['clone'] + args + [self.repourl, '.']
+            command += ['--reference', self.reference]
+        if self.origin:
+            command += ['--origin', self.origin]
+        command += [self.repourl, '.']
```

Comment:
Note, the change above is mostly non-functional except for the origin line. This removes the variable 'args' and uses 'command' only. I think this improves readability.